### DISCUSSION TITLE
I get lost almost every day and don't know which file I'm editing.

### DISCUSSION THREAD

**Discussion by vinciarts** [godotengine/godot-proposals#6339] [GUI]
Hi, I've been using Godot for about 1 week. I still don't know what these highlight mean.
Whenever I want to find a file to edit, it costs me a lot of time and energy and made me a lot of mistakes.

![image](https://user-images.githubusercontent.com/891308/220678382-6c2b2706-aaec-49e6-b233-96a70fc296d0.png)

My suggestion:

<img width="498" alt="image" src="https://user-images.githubusercontent.com/891308/220683687-f6b2c2ba-da62-4be0-9465-53c1952d504e.png">

**Comment by AThousandShips** [comment]
What confuses you? The highlight for "map" is simply the current tab in the editor, and the one in the script editor is for the scripts, what changes do you suggest? Why is the current system bad?

**Reply by CarpenterBlue** [reply]
Editor sometimes f*cks up after you make a new script resource and shows you wrong script file in the code editor.
That definitely confuses me sometimes. I remedied the situation by some proactivity but it is still a major issue.

**Reply by AThousandShips** [reply]
Please open an issue for this, but this isn't the discussion for that

**Comment by AThousandShips** [comment]
As I am not sure what it is you are asking for I don't know if this is at all about what you're having an issue with, but I will try answer one part:
The reason the FileSystem viewer doesn't show which file is active is probably because there's no easy answer to "which file is open", you can have a scene open, and a script, and a resource in the inspector, at least two of them at once, if you are needing to find the file you can right click the scene tab and press the option "Show In File System" to go to the file there.

**Comment by timothyqiu** [comment]
FileSystem is an independent dock. It's main purpose is just for opening files quickly.

Among all the file types that can be edited, the most common one is the `.tscn` file which represents a scene. Once a scene is opened:

* It's add as a tab at the top of the main area. You can switch between opened scenes there.
* Scene dock displays the node structure of the current edited scene.

You can find four buttons at the top of the screen: 2D, 3D, Script, and AssetLib. 2D and 3D are different views of the current scene. In Godot, 2D is not implemented as a special mode in the 3D space. 2D and 3D elements can't see each other, thus the two separate views.

The Script button switches you to the builtin script editor. It manages its own list of opened scripts. It's not related to any scene. But if the root node of the scene have a script attached, the script will be opened by default when you open the scene.

It's indeed a bit confusing for new users. Because the top buttons and scene tabs are layouted in an counterintuitive way. I remember there were already several proposals / discussions about this.
